spec/mongo/socket/ssl_spec.rb

```diff
@@ -279,6 +279,42 @@
       end
     end
 
+    context 'when the hostname is incorrect' do
+      let(:host_name) do
+        'incorrect_hostname'
+      end
+
+      context 'when the hostname is verified' do
+        it 'raises an error' do
+          expect {
+            described_class.new(
+              address.host,
+              address.port,
+              host_name,
+              30,
+              ::Socket::PF_INET,
+              options.merge(tls_allow_invalid_hostnames: false)
+            ).connect!
+          }.to raise_error(Mongo::Error::SocketError)
+        end
+      end
+
+      context 'when the hostname is not verified' do
+        it 'raises an error' do
+          expect {
+            described_class.new(
+              address.host,
+              address.port,
+              host_name,
+              30,
+              ::Socket::PF_INET,
+              options.merge(tls_allow_invalid_hostnames: true, ssl_verify: false)
```

Comment:
Where does https://github.com/mongodb/specifications/blob/master/source/uri-options/uri-options.rst say what should happen when these two options are both provided?